lib/sycamore/sycamore/functions/tokenizer.py

```diff
@@ -1,28 +1,41 @@
-import tiktoken
-from transformers import AutoTokenizer
+from abc import ABC, abstractmethod
+from typing import Union
 
 
-class Tokenizer:
-    def tokenize(self, text: str):
+class Tokenizer(ABC):
+    @abstractmethod
+    def tokenize(self, text: str, as_ints: bool = False) -> Union[list[int], list[str]]:
         pass
 
 
 class OpenAITokenizer(Tokenizer):
     def __init__(self, model_name: str):
+        import tiktoken
+
         self._tk = tiktoken.encoding_for_model(model_name)
 
-    def tokenize(self, text: str):
-        return self._tk.encode(text)
+    def tokenize(self, text: str, as_ints: bool = False):
+        token_ids = self._tk.encode(text)
+        if as_ints:
+            return token_ids
+        tokens = self._tk.decode_batch([[id] for id in token_ids])
```

Comment:
What's the plus/minus on having two methods (with clear signatures) versus this one method with a behavior flag?

Reply:
The previous implementations of the tokenizers did different things so I went with this interface